src/java/com/rapleaf/jack/queries/IndexHints.java

```diff
@@ -5,52 +5,52 @@
   private IndexHints() {
   }
 
-  public static IndexHint use(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ALL, index, indices);
+  public static IndexHint use(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ALL, indices);
   }
 
-  public static IndexHint useForJoin(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.JOIN, index, indices);
+  public static IndexHint useForJoin(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.JOIN, indices);
   }
 
-  public static IndexHint useForOrderBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ORDER_BY, index, indices);
+  public static IndexHint useForOrderBy(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.ORDER_BY, indices);
   }
 
-  public static IndexHint useForGroupBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.GROUP_BY, index, indices);
+  public static IndexHint useForGroupBy(Index... indices) {
+    return new IndexHint(IndexHint.Type.USE, IndexHint.Scope.GROUP_BY, indices);
   }
 
-  public static IndexHint force(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ALL, index, indices);
+  public static IndexHint force(Index... indices) {
+    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ALL, indices);
   }
 
-  public static IndexHint forceForJoin(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.JOIN, index, indices);
+  public static IndexHint forceForJoin(Index... indices) {
+    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.JOIN, indices);
   }
 
-  public static IndexHint forceForOrderBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ORDER_BY, index, indices);
+  public static IndexHint forceForOrderBy(Index... indices) {
+    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.ORDER_BY, indices);
   }
 
-  public static IndexHint forceForGroupBy(Index index, Index... indices) {
-    return new IndexHint(IndexHint.Type.FORCE, IndexHint.Scope.GROUP_BY, index, indices);
+  public static IndexHint forceForGroupBy(Index... indices) {
```

Comment:
I prefer to keep the `T t, T... ts` pattern. True that in `In / NotIn` it does not prevent runtime exceptions due to empty collections, but in other class it does. In addition the code has already been written, so the cost of the complication has already been paid.